uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/instances/leave-request.instance.js

```diff
@@ -11,6 +11,71 @@ define([
   instances.factory('LeaveRequestInstance', ['$q', 'checkPermissions', 'OptionGroup',
     'shared-settings', 'ModelInstance', 'LeaveRequestAPI',
     function ($q, checkPermissions, OptionGroup, sharedSettings, ModelInstance, LeaveRequestAPI) {
+      /**
+       * Amends balance change breakdown if calculation unit absence type is "hours".
+       * It changes the balance first and last days (only first day in case of single day request)
```

Comment:
i still have no idea what this means, neither by reading the description or by looking at the code. Could you explain it to me?

Reply:
Balance change API returns an object containing total amount and an array of days with separate amounts. For "hours" unit you must ignore first and last day, it always goes from the deductions user selected. The amount also should be calculated based on that. So the balance object needs to be "amended" for hours type.

Reply:
i'll wait for the both functions to have proper names and the documentation to be more descriptive

> So the balance object needs to be "amended" for hours type.

Simply have the functions return the new value, the code would be easier to follow and we would not have to leave "@NOTE"s around

Reply:
done.